src/hoa-convolver.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileOverview A collection of convolvers. Can be used for the optimized HOA
+ *               binaural rendering. (e.g. SH-MaxRe HRTFs)
+ */
+
+'use strict';
+
+/**
+ * @class HOAConvolver
+ * @description A collection of convolvers for N-channel HOA stream.
+ * @param {AudioContext} context          Associated AudioContext.
+ * @param {Object} options                Options for speaker.
+ * @param {Array} options.IR              Array of buffers for HRTF convolution.
+ * @param {Number} options.ambisonicOrder Ambisonic order (default is 3).
+ * @param {Number} options.gain           Post-gain for the speaker (optional).
+ */
+function HOAConvolver (context, options) {
+  this._active = false;
+
+  this._context = context;
+
+  var ambisonicOrder = 3;
+  if (options.ambisonicOrder) {
+    ambisonicOrder = options.ambisonicOrder;
+  }
+  var num_channels = (ambisonicOrder + 1) * (ambisonicOrder + 1);
+  this._input = this._context.createChannelSplitter(num_channels);
+
+  this._symmetrics = this._context.createGain();
+  this._antisymmetrics = this._context.createGain();
+
+  this._inverter = this._context.createGain();
+  this._inverter.gain.value = -1;
+
+  this._binaural = this._context.createChannelMerger(2);
+  this._outputGain = this._context.createGain();
+  if (options.gain) {
+    this._outputGain.gain.value = options.gain;
+  }
+
+  this._convolvers = new Array(num_channels);
```

Comment:
A convolver can take a stereo IR and its efficiency is better than 2 mono convolvers. That's how FOA is implemented. It would be great if you can describe how many IR files and convolvers (mono/stereo) are being used here.